README.md

```diff
@@ -591,6 +609,214 @@ ObxValue((data) => Switch(
+
+```dart
+ class AwesomeController extends GetxController {
+   final String title = 'My Awesome View';
+ }
+  
+  // ALWAYS remember to pass the `Type` you used to register your controller!
+ class AwesomeView extends GetView<AwesomeController> {
+   @override
+   Widget build(BuildContext context) {
+     return Container(
+       padding: EdgeInsets.all(20),
+       child: Text( controller.title ), // just call `controller.something`
+     );
+   }
+ }
+```
+
+#### GetWidget
+
+Most people have no idea about this Widget, or totally confuse the usage of it.
+The use case is very rare, but very specific: It `caches` a Controller.
+Because of the _cache_, can't be a `const Stateless`.
+
+> So, when do you need to "cache" a Controller?
+
+If you use, another "not so common" feature of **GetX**: `Get.create()`.
+
+`Get.create(()=>Controller())` will generate a new `Controller` each time you call
+`Get.find<Controller>()`, 
+
+That's where `GetWidget` shines... as you can use it, for example,
+to keep a list of Todo items. So, if the widget gets "rebuilt", it will keep the same controller instance.
+
+
+#### GetxService
+
+This class is like a `GetxController`, it shares the same lifecycle ( `onInit()`, `onReady()`, `onClose()`).
+But has no "logic" inside of it. It just notifies **GetX** Dependency Injection system, that this subclass
+**can not** be removed from memory. 
+
+So is super useful to keep your "Services" always reachable and active with `Get.find()`. Like:
+`ApiService`, `StorageService`, `CacheService`. 
+
+```dart
+Future<void> main() async {
+  await initServices(); /// AWAIT SERVICES INITIALIZATION.
+  runApp(SomeApp());
+}
+
+/// Is a smart move to make your Services intiialize before you run the Flutter app.
+/// as you can control the execution flow (maybe you need to load some Theme configuration, 
+/// apiKey, language defined by the User... so load SettingService before running ApiService.
+/// so GetMaterialApp() doesnt have to rebuild, and takes the values directly.    
+void initServices() async {
+  print('starting services ...');
+  /// Here is where you put get_storage, hive, shared_pref initialization.
+  /// or moor connection, or whatever that's async.
+  await Get.putAsync(() => DbService().init());
+  await Get.put(SettingsService()).init();
+  print('All services started...');
+}
+
+class DbService extends GetxService {
+  Future<DbService> init() async {
+    print('$runtimeType delays 2 sec');
+    await 2.delay();
+    print('$runtimeType ready!');
+    return this;
+  }
+}
+
+class SettingsService extends GetxService {
+  void init() async {
+    print('$runtimeType delays 2 sec');
+    await 1.delay();
+    print('$runtimeType ready!');
+  }
+}
+
+```
+
+The only way to actually delete a `GetxService`, is with `Get.reset()` which is like a
+"Hot Reboot" of your app. So remember, if you need absolute persistance of a class instance during the
+lifetime of your app, use `GetxService`. 
+  
+
+ 
+
+
```

Comment:
the markdown ignores by default more than one line, but even so, it's a good practice to not let more than 1 blank line